src/commands/delete.js

```diff
@@ -0,0 +1,53 @@
+const utils = require('../utils');
+
+const _delete = (context, appOrVersion, version) => {
+  if (appOrVersion === 'version' && !version) {
+    const message = 'Error: No version - provide a version like "1.0.0"...';
+    return Promise.reject(new Error(message));
+  }
+  return utils.checkCredentials()
+    .then(() => utils.getLinkedApp())
+    .then((app) => {
+      const deletePrepMessage = (appOrVersion === 'version')
```

Comment:
I'd also argue that this could be moved out and separated. E.g.

```js
return isDeletingVersion ? deleteVersion() : deleteApp();
```

I'm not as familiar with the code, but I would have expected an `App` model that would have these methods/actions. That would be a non-blocker, though, if it's not the norm.